src/libcommon/log/sentry/handler.cpp

```diff
@@ -434,7 +446,158 @@ SentryHandler::~SentryHandler() {
+        stopPTrace(childPerformanceTraceId2, aborted);
+    }
+
+    // Stop the PerformanceTrace
+    if (performanceTrace.isSpan()) {
+        if (!performanceTrace.span()) {
+            assert(false && "Span is not valid");
+            return;
+        }
+        if (aborted) sentry_span_set_status(performanceTrace.span(), SENTRY_SPAN_STATUS_ABORTED);
+        sentry_span_finish(performanceTrace.span()); // Automatically stop any child transaction
+    } else {
+        if (!performanceTrace.transaction()) {
+            assert(false && "Transaction is not valid");
+            return;
+        }
+        if (aborted)
+            sentry_transaction_set_status(performanceTrace.transaction(),
+                                          SENTRY_SPAN_STATUS_ABORTED /*This enum is also valid for a transactrion.*/);
+        sentry_transaction_finish(performanceTrace.transaction()); // Automatically stop any child transaction
+    }
+    _performanceTraces.erase(id);
+}
+
+pTraceId Handler::startTransaction(const std::string &name, const std::string &description) {
+    sentry_transaction_context_t *tx_ctx = sentry_transaction_context_new(name.c_str(), description.c_str());
+    sentry_transaction_t *tx = sentry_transaction_start(tx_ctx, sentry_value_new_null());
+    _pTraceIdCounter++;
+    pTraceId traceId = _pTraceIdCounter;
+    auto [it, res] = _performanceTraces.try_emplace(traceId, traceId);
+    if (!res) {
+        assert(false && "Transaction already exists");
+        return 0;
+    }
+
+    PerformanceTrace &pTrace = it->second;
+    pTrace.setTransaction(tx);
+
+    return _pTraceIdCounter;
+}
+
+pTraceId Handler::startSpan(const std::string &name, const std::string &description, const pTraceId &parentId) {
+    auto parentIt = _performanceTraces.find(parentId);
+    if (parentIt == _performanceTraces.end()) {
+        assert(false && "Parent transaction/span does not exist");
+        return 0;
+    }
+
+    PerformanceTrace &parent = parentIt->second;
+    sentry_span_t *span = nullptr;
+    if (parent.isSpan()) {
+        if (!parent.span()) {
+            assert(false && "Parent span is not valid");
+            return 0;
+        }
+        span = sentry_span_start_child(parent.span(), name.c_str(), description.c_str());
+        assert(span);
+    } else {
+        if (!parent.transaction()) {
+            assert(false && "Parent transaction is not valid");
+            return 0;
+        }
+        span = sentry_transaction_start_child(parent.transaction(), name.c_str(), description.c_str());
+        assert(span);
+    }
+
+    _pTraceIdCounter++;
+    pTraceId traceId = _pTraceIdCounter;
+    auto [it, res] = _performanceTraces.try_emplace(traceId, traceId);
+    if (!res) {
+        assert(false && "Transaction already exists");
+        return 0;
+    }
+
+    PerformanceTrace &pTrace = it->second;
+    pTrace.setSpan(span);
+    parent.addChild(pTrace.id());
+    return traceId;
```

Comment:
So, we return `_pTraceIdCounter++`?

Reply:
Humm, yes. Not sure to understand your point 😅

Reply:
Sorry, it was not clear: it seems that `traceId` is just an alias for ` _pTraceIdCounter`.  Why using two variables to store the same value?

Maybe this pattern would make things clearer:
`pTraceId traceId = makeUniqueTraceId()` and `pTraceId  makeUniqueTraceId() { return _pTraceCounter++; }`.

Reply:
Done here: [c696656](https://github.com/Infomaniak/desktop-kDrive/pull/403/commits/c696656d1d9c47f586efbeb49831dd6a5c23d640)